packages/ember-metal/lib/injected_property.js

```diff
@@ -35,5 +36,10 @@ function injectedPropertyGet(keyName) {
   assert(`InjectedProperties should be defined with the inject computed property macros.`, desc && desc.type);
   assert(`Attempting to lookup an injected property on an object without a container, ensure that the object was instantiated via a container.`, owner);
 
-  return owner.lookup(`${desc.type}:${desc.name || keyName}`);
+  let specifier = `${desc.type}:${desc.name || keyName}`;
+  if (desc.source) {
```

Comment:
I don’t fully grok why we need this conditional. Seems to be that calling lookup like `owner.lookup(specifier, { source: undefined })` (which is what would happen if `desc.source` is not present) would be fine. What am I missing?